### PR TITLE
refactor: support local install with npm 5

### PR DESCRIPTION
**Pull request by sis0k0**
**NPM 5+ creates symlinks for all locally installed node packages.**

- use INIT_CWD instead of __dirname when the package is symlinked
`process.env.INIT_CWD` is present from npm 5.4 and contains the original working directory that npm was executed from, opposed to __dirname which contains the script directory (if it's a symlink - the original destination on the file system).
If we installed nativescript-dev-webpack from `/home/nativescript-dev-webpack` with npm 5+, a symlink will be created. Then, if we try to invoke `ns-bundle`:
``` js
// /home/project/node_modules/nativescript-dev-webpack/bin/ns-bundle

console.log(process.env.INIT_CWD); // /home/project
console.log(__dirname); // /home/nativescript-dev-webpack/bin
```

Since the script needs the project directory, we need to use `INIT_CWD` instead of `__dirname`.

- ns-bundle script: invoke webpack with node and --preserve-symlinks
```
node --preserve-symlinks ./node_modules/.bin/webpack
```
By default, node resolves all symlinked packages to their original locations. For example, if your project directory is `/home/my-project` and you installed `/home/nativescript-dev-webpack`, node will resolve all imports of `nativescript-dev-webpack` from its original location - `/home/nativescript-dev-webpack`.
This will cause the following import (one of many) to fail:
``` js
// nativescript-dev-webpack/plugins/GenerateBundleStarterPlugin.js
const { RawSource } = require("webpack-sources");
``` 
The `webpack-sources` package is not a dependency of the `nativescript-dev-webpack` plugin and therefore, is not present in `/home/nativescript-dev-webpack/node_modules`. However, it is a dependency of the project and is meant to be resolved from there (`/home/project/node_modules/webpack-sources`).
With this change, when we are spawning new child process for webpack from the `ns-bundle` script, we provide the `--preserve-symlinks` flag to instruct node to preserve all symlinks, therefore the plugin will be resolved from `/project/node_modules/nativescript-dev-webpack`.

### CODE REVIEW

**Comment by sis0k0** [comment]
run ci

**Comment by sis0k0** [comment]
run ci

**Comment by vchimev** [comment]
run ci

File: projectHelpers.js [hunk 1]
```diff
@@ -30,11 +30,36 @@ const writePackageJson = (content, projectDir) => {
     fs.writeFileSync(packageJsonPath, JSON.stringify(content, null, 2))
 }
 
+const getProjectDir = ({ nestingLvl } = { nestingLvl: 0 }) => {
```
Comment (PanayotCankov) [inline projectHelpers.js]: This may be useful: https://www.npmjs.com/package/app-root-path or may be not. :)

File: bin/ns-bundle [hunk 2]
```diff
@@ -152,7 +152,9 @@ function webpack(platform, env) {
         console.log(`Running webpack for ${platform}...`);
 
         const args = [
-            `webpack`,
+            `node`,
+            `--preserve-symlinks`,
```
Comment (PanayotCankov) [inline bin/ns-bundle]: 👍

**Pull request merged**